### ISSUE TITLE
Bug in set_symmetric_diffference logic

### ISSUE THREAD

**Issue by 0racle** [rakudo/rakudo#4118]
## The Problem

When reducing multiple lists with the symmetric difference operator, items present in more than one list should not exist in the result.

```perl6
my @a = (    1, 2,      );
my @b = ( 0, 1, 2, 3, 4 );
my @c = ( 0,    2,    4 );

say [⊖] @a, @b, @c;  # Set(0, 3)
```

## Expected Behavior

```perl-6
say [⊖] (1,2,3), (1,2,3), (1,2,3);  # Set()
say [⊖] (0,1,2), (0,1,2), (0,1,2);  # Set()
```

## Actual Behavior

```perl-6
say [⊖] (1,2,3), (1,2,3), (1,2,3);  # Set()
say [⊖] (0,1,2), (0,1,2), (0,1,2);  # Set(0)
```

## Steps to Reproduce

The trigger seems to be numeric `0` in the Lists. String `"0"` is not affected. This leads me to suspect the root cause is the logic is doing a truthiness check rather than a definedness check.

Bisectable found  [this](https://github.com/rakudo/rakudo/commit/0c02f93e327617db04c804dfa72f00dcf273f6f8) commit.

**Comment by lizmat** [comment]
@Altai-man this is *not* a blocker, but will try to fix before 2020.12 release nonetheless

**Comment by melezhik** [comment]
Added to RICH - http://rakudist.raku.org/sparky/report/RakuPlay-2/6327
